### ISSUE TITLE
A lot of OutOfMemoryError exceptions

### ISSUE THREAD

**Issue by gholz** [Pushwoosh/pushwoosh-android-sdk#103]
This is not a new issue but the previous one was closed by some reason. We're still getting a lot of this and it's been like this for months now.

```
Package: com.getyourguide.android
Version Code: 252000003
Version Name: 2.52.3
Android: 7.0
Android Build: RENOVATE ICE G95X 6.0
Manufacturer: samsung
Model: SM-G950F
Thread: main-1
CrashReporter Key: 1C9725A5-45CC-A833-11F7-3D91E824F63144FBD81C
Start Date: Fri Feb 02 17:59:49 GMT+01:00 2018
Date: Fri Feb 02 18:01:23 GMT+01:00 2018

java.lang.OutOfMemoryError: Failed to allocate a 131084 byte allocation with 6443864 free bytes and 6MB until OOM; failed due to fragmentation (required continguous free 135168 bytes where largest contiguous free 77824 bytes)
	at java.util.HashMap.inflateTable(HashMap.java:287)
	at java.util.HashMap.<init>(HashMap.java:255)
	at android.app.SharedPreferencesImpl$EditorImpl.commitToMemory(SharedPreferencesImpl.java:402)
	at android.app.SharedPreferencesImpl$EditorImpl.apply(SharedPreferencesImpl.java:362)
	at com.pushwoosh.a.d.a(Unknown Source)
	at com.pushwoosh.notification.LocalNotificationReceiver.scheduleNotification(Unknown Source)
	at com.pushwoosh.notification.d.e(Unknown Source)
	at com.pushwoosh.c.a(Unknown Source)
	at com.pushwoosh.c.bridge$lambda$2$c(Unknown Source)
	at com.pushwoosh.c$$Lambda$2.onReceive(Unknown Source)
	at com.pushwoosh.internal.event.EventBus.a(Unknown Source)
	at com.pushwoosh.internal.event.EventBus.b(Unknown Source)
	at com.pushwoosh.internal.event.EventBus.bridge$lambda$0$EventBus(Unknown Source)
	at com.pushwoosh.internal.event.EventBus$$Lambda$0.run(Unknown Source)
	at android.os.Handler.handleCallback(Handler.java:751)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6776)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1520)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1410)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:107)
```

**Comment by wfhm** [comment]
Hi,

Are you referring to this one - https://github.com/Pushwoosh/pushwoosh-android-sdk/issues/62? It was fixed with the 5.4.1 release, however the changes were rolled back shortly after, and, as we have received another report in a separate thread, we kept only the last one opened.

The issue itself requires storage refactoring, which is already planned, however right now I cannot provide you with any particular ETA. Once there is any news we will update you via this thread.
